exercises/Stephen_Smith/ex01_basics/src/main.cpp

```diff
@@ -0,0 +1,151 @@
+/**
+ * Skeleton main routine
+ */
+
+
+#include <iostream>
+#include <fstream>
+#include <map>
+#include <iomanip>
+#include "../include/splitLineIntoWords.h"
+
+
+
+
+
+namespace{
+    const int MIN_WORD_LENGTH=5;
+}
+
+// Remove punctuation from the input word
+int removePunctuation(std::string &iString)
+{
+
+    std::string::iterator itr = iString.begin() ;
+
+    static const std::string punctuation(".!,\"?!");
+
+    while (itr != iString.end())
+    {
+
+        char c = *itr;
+        if ( punctuation.find(c) != std::string::npos )
+        {
+            itr = iString.erase(itr);
+        }else{
+            ++itr;
+        }
+
+    }
+
+    return 0;
+
+}
+
+// Function to compare counts of two words eached stored in std::pair with the associated count
+// Input paramter = std::pair of <word,count>
+bool compareWordCounts(std::pair<std::string,int> a,std::pair<std::string,int> b)
```

Comment:
As `std::pair<std::string, int>` is used in a few places then maybe consider defining an alias above the anonymous namespace as

using WordCount = std::pair<std::string, int>;